### DISCUSSION TITLE
Feature Request - Set Vibrate Duration Per Alarm

### DISCUSSION THREAD

**Discussion by erobinson** [NightscoutFoundation/xDrip#2099] [Ideas]
Hi! I think it would be awesome to be able to set short, medium, or long vibrations per alert. For example, I would love a short nudge if my BG is over 130, a medium vibration if over 160, long vibration if over 200, and double long vibration if below 70.

The primary thing is that I would love to have a small notification if over something 130 and trending up but don't want the current loud/long vibration.

**Comment by Navid200** [comment]
Please have a look at this and see if it is the same or not:
https://github.com/NightscoutFoundation/xDrip/issues/1448

**Reply by erobinson** [reply]
Thanks! That is exactly it. I created a PR https://github.com/NightscoutFoundation/xDrip/pull/2142

**Reply by Navid200** [reply]
That's great.

I cannot guarantee that it will be merged if you do what I suggest.

I would add a description explaining what it does.
For any new settings, I would include a screenshot so that the reviewer can see exactly what it will look like.
Unless it is very simple, I would also explain how I tested it.
If you believe that your PR fixes an issue, you can include the following in the description:  "Fixes link" without the quotes, where link would be an absolute link to the issue.  This will create a visible link for everyone between the PR and the issue.  Visible even looking at all PRs without descending inside the PR.

But, as I said, I am not one who can approve it.

**Reply by erobinson** [reply]
👍 Will add what you suggest 😄 There is currently a bug, so still need to do some more testing on it. Will also refactor it to use strings & add some additional vibration patterns.

**Reply by Navid200** [reply]
You can convert it to a draft until it is ready for review.
